// rnv/dist/engine-core/commonEngine.js
Object.defineProperty(exports,"__esModule",{value:true});exports.getPlatformProjectDir=exports.getTemplateProjectDir=exports.ejectPlatform=exports.getTemplateRootDir=exports.getPlatformOutputDir=exports.getPlatformBuildDir=void 0;var _common=require("../core/common");
var _logger=require("../core/systemManager/logger");

var getPlatformBuildDir=function getPlatformBuildDir(c){return(0,_common.getAppFolder)(c);};exports.getPlatformBuildDir=getPlatformBuildDir;

var getPlatformOutputDir=function getPlatformOutputDir(){
(0,_logger.logError)('core engine does not support getPlatformOutputDir');
return null;
};exports.getPlatformOutputDir=getPlatformOutputDir;

var getTemplateRootDir=function getTemplateRootDir(){
(0,_logger.logError)('core engine does not support getTemplateRootDir');
return null;
};exports.getTemplateRootDir=getTemplateRootDir;

var ejectPlatform=function ejectPlatform(){
(0,_logger.logError)('core engine does not support ejectPlatform');

};exports.ejectPlatform=ejectPlatform;

var getTemplateProjectDir=function getTemplateProjectDir(c){
var dir=(0,_common.getTemplateDir)(c);
var output;
switch(c.platform){
default:
output=dir;}

return output;
};exports.getTemplateProjectDir=getTemplateProjectDir;

var getPlatformProjectDir=function getPlatformProjectDir(c){
var dir=getPlatformBuildDir(c);
var output;
switch(c.platform){
default:
output=dir;}

return output;
};exports.getPlatformProjectDir=getPlatformProjectDir;
