// rnv/dist/systemTools/objectUtils.js
Object.defineProperty(exports,"__esModule",{value:true});exports.isSymbol=exports.isDate=exports.isError=exports.isRegExp=exports.isLikeNull=exports.isUndefined=exports.isNull=exports.isBool=exports.isFunction=exports.isNumber=exports.isString=exports.isArray=exports.isObject=void 0;
var isObject=function isObject(value){return value&&typeof value==='object'&&value.constructor===Object;};exports.isObject=isObject;

var isArray=function isArray(value){return value&&typeof value==='object'&&value.constructor===Array;};exports.isArray=isArray;

var isString=function isString(value){return typeof value==='string'||value instanceof String;};exports.isString=isString;

var isNumber=function isNumber(value){return typeof value==='number'&&Number.isFinite(value);};exports.isNumber=isNumber;

var isFunction=function isFunction(value){return typeof value==='function';};exports.isFunction=isFunction;

var isBool=function isBool(value){return typeof value==='boolean';};exports.isBool=isBool;

var isNull=function isNull(value){return value===null;};exports.isNull=isNull;

var isUndefined=function isUndefined(value){return typeof value==='undefined';};exports.isUndefined=isUndefined;

var isRegExp=function isRegExp(value){return value&&typeof value==='object'&&value.constructor===RegExp;};exports.isRegExp=isRegExp;

var isError=function isError(value){return value instanceof Error&&typeof value.message!=='undefined';};exports.isError=isError;

var isDate=function isDate(value){return value instanceof Date;};exports.isDate=isDate;

var isSymbol=function isSymbol(value){return typeof value==='symbol';};exports.isSymbol=isSymbol;

var isLikeNull=function isLikeNull(value){return isNull(value)||isUndefined(value);};exports.isLikeNull=isLikeNull;
