// iocane/dist/node/defaults.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var derivation_1 = require("./derivation");
var encryption_1 = require("../node/encryption");
var packing_1 = require("./packing");
var shared_1 = require("../base/shared");
var SALT_LENGTH = 12;
/**
 * Get the default options
 * @returns Default configuration options
 */
function getDefaultOptions() {
    return {
        decryption_cbc: encryption_1.decryptCBC,
        decryption_gcm: encryption_1.decryptGCM,
        derivationRounds: shared_1.DERIVED_KEY_ITERATIONS,
        deriveKey: derivation_1.deriveFromPassword,
        encryption_cbc: encryption_1.encryptCBC,
        encryption_gcm: encryption_1.encryptGCM,
        generateIV: encryption_1.generateIV,
        generateSalt: encryption_1.generateSalt,
        method: shared_1.ALGO_DEFAULT,
        pack_data: packing_1.packEncryptedData,
        pack_text: packing_1.packEncryptedText,
        pbkdf2: derivation_1.pbkdf2,
        saltLength: SALT_LENGTH,
        unpack_data: packing_1.unpackEncryptedData,
        unpack_text: packing_1.unpackEncryptedText
    };
}
exports.getDefaultOptions = getDefaultOptions;
