// rnv/dist/core/constants.js
var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.RN_CLI_CONFIG_NAME=exports.RENATIVE_CONFIG_ENGINES_NAME=exports.RENATIVE_CONFIG_PLATFORMS_NAME=exports.RENATIVE_CONFIG_TEMPLATES_NAME=exports.RENATIVE_CONFIG_PLUGINS_NAME=exports.RENATIVE_CONFIG_WORKSPACES_NAME=exports.RENATIVE_CONFIG_RUNTIME_NAME=exports.RENATIVE_CONFIG_BUILD_NAME=exports.RENATIVE_CONFIG_TEMPLATE_NAME=exports.RENATIVE_CONFIG_PRIVATE_NAME=exports.RENATIVE_CONFIG_LOCAL_NAME=exports.RENATIVE_CONFIG_NAME=exports.KAIOS_SDK=exports.WEBOS_SDK=exports.TIZEN_SDK=exports.ANDROID_NDK=exports.ANDROID_SDK=exports.CLI_WEBOS_ARES_NOVACOM=exports.CLI_WEBOS_ARES_DEVICE_INFO=exports.CLI_WEBOS_ARES_SETUP_DEVICE=exports.CLI_WEBOS_ARES_LAUNCH=exports.CLI_WEBOS_ARES_INSTALL=exports.CLI_WEBOS_ARES_PACKAGE=exports.CLI_WEBOS_ARES=exports.CLI_SDB_TIZEN=exports.CLI_TIZEN=exports.CLI_KAIOS_EMULATOR=exports.CLI_TIZEN_EMULATOR=exports.CLI_ANDROID_SDKMANAGER=exports.CLI_ANDROID_AVDMANAGER=exports.CLI_ANDROID_ADB=exports.CLI_ANDROID_EMULATOR=exports.FORM_FACTOR_TV=exports.FORM_FACTOR_WATCH=exports.FORM_FACTOR_DESKTOP=exports.FORM_FACTOR_MOBILE=exports.PLATFORM_GROUP_ELECTRON=exports.PLATFORM_GROUP_SMARTTV=exports.WEB_HOSTED_PLATFORMS=exports.EXTENSIONS=exports.PLATFORMS=exports.REMOTE_DEBUG_PORT=exports.OS=exports.OS_LINUX=exports.OS_MAC=exports.OS_WINDOWS=exports.ICONS=exports.XBOX360=exports.XBOX=exports.WP8=exports.WP10=exports.WINDOWS=exports.WII=exports.WEBIAN=exports.WEBOS=exports.WEB=exports.WATCHOS=exports.VIZIO=exports.VIERACONNECT=exports.VIDAA=exports.VEWD=exports.UNITY=exports.UBUNTU_TOUCH=exports.UBUNTU=exports.TVOS=exports.TIZEN_MOBILE=exports.TIZEN_WATCH=exports.TIZEN=exports.TIVO=exports.SAILFISH=exports.ROKU=exports.PS4=exports.ORSAY=exports.OCCULUS=exports.NETCAST=exports.MEEGO=exports.MACOS=exports.KAIOS=exports.IOS=exports.HBBTV=exports.FIRE_TV=exports.FIRE_OS=exports.FIREFOX_TV=exports.FIREFOX_OS=exports.CHROME_OS=exports.CHROMECAST=exports.BLACKBERRY=exports.ASTIAN=exports.APPLE_AUTO=exports.ALEXA=exports.ANDROID_WEAR=exports.ANDROID_TV=exports.ANDROID_AUTO=exports.ANDROID=exports.RNV_SERVER_DIR_NAME=exports.RNV_PROJECT_DIR_NAME=exports.RNV_NODE_MODULES_DIR=exports.CURRENT_DIR=exports.RNV_HOME_DIR=exports.USER_HOME_DIR=void 0;exports.REMOTE_DEBUGGER_ENABLED_PLATFORMS=exports.REDASH_KEY=exports.REDASH_URL=exports.INJECTABLE_RUNTIME_PROPS=exports.INJECTABLE_CONFIG_PROPS=exports.configSchema=exports.PARAMS=exports.TASK_KILL=exports.TASK_CONFIGURE_SOFT=exports.TASK_WORKSPACE_CONFIGURE=exports.TASK_APP_CREATE=exports.TASK_APP_CONFIGURE=exports.TASK_PKG=exports.TASK_HOOKS_PIPES=exports.TASK_HOOKS_LIST=exports.TASK_HOOKS_RUN=exports.TASK_CRYPTO_UPDATE_PROFILES=exports.TASK_CRYPTO_UPDATE_PROFILE=exports.TASK_CRYPTO_INSTALL_PROFILE=exports.TASK_CRYPTO_INSTALL_PROFILES=exports.TASK_CRYPTO_INSTALL_CERTS=exports.TASK_CRYPTO_DECRYPT=exports.TASK_CRYPTO_ENCRYPT=exports.TASK_PLUGIN_UPDATE=exports.TASK_PLUGIN_LIST=exports.TASK_PLUGIN_ADD=exports.TASK_PROJECT_UPGRADE=exports.TASK_PROJECT_CONFIGURE=exports.TASK_PLATFORM_SETUP=exports.TASK_PLATFORM_LIST=exports.TASK_PLATFORM_EJECT=exports.TASK_PLATFORM_CONNECT=exports.TASK_PLATFORM_CONFIGURE=exports.TASK_WORKSPACE_UPDATE=exports.TASK_WORKSPACE_LIST=exports.TASK_WORKSPACE_CONNECT=exports.TASK_WORKSPACE_ADD=exports.TASK_TEMPLATE_APPLY=exports.TASK_TEMPLATE_LIST=exports.TASK_TEMPLATE_ADD=exports.TASK_TARGET=exports.TASK_TARGET_LIST=exports.TASK_TARGET_LAUNCH=exports.TASK_SWITCH=exports.TASK_STATUS=exports.TASK_PUBLISH=exports.TASK_INSTALL=exports.TASK_CLEAN=exports.TASK_LOG=exports.TASK_DEPLOY=exports.TASK_PACKAGE=exports.TASK_DEBUG=exports.TASK_EXPORT=exports.TASK_START=exports.TASK_INFO=exports.TASK_BUILD=exports.TASK_DOCTOR=exports.TASK_CONFIGURE=exports.TASK_RUN=exports.SDK_PLATFORMS=exports.SUPPORTED_PLATFORMS_LINUX=exports.SUPPORTED_PLATFORMS_WIN=exports.SUPPORTED_PLATFORMS_MAC=exports.SUPPORTED_PLATFORMS=exports.PACKAGE_JSON_FILEDS=exports.IS_WEAR_UNDER_SIZE=exports.IS_TABLET_ABOVE_INCH=exports.SAMPLE_APP_ID=exports.NEXT_CONFIG_NAME=exports.RN_BABEL_CONFIG_NAME=void 0;var _toConsumableArray2=_interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _os=require("os");
var _path=_interopRequireDefault(require("path"));

var USER_HOME_DIR=(0,_os.homedir)();exports.USER_HOME_DIR=USER_HOME_DIR;
var RNV_HOME_DIR=_path.default.join(__dirname,'../..');exports.RNV_HOME_DIR=RNV_HOME_DIR;
var CURRENT_DIR=_path.default.resolve('.');exports.CURRENT_DIR=CURRENT_DIR;
var RNV_NODE_MODULES_DIR=_path.default.join(RNV_HOME_DIR,'node_modules');exports.RNV_NODE_MODULES_DIR=RNV_NODE_MODULES_DIR;
var RNV_PROJECT_DIR_NAME='project';exports.RNV_PROJECT_DIR_NAME=RNV_PROJECT_DIR_NAME;
var RNV_SERVER_DIR_NAME='server';exports.RNV_SERVER_DIR_NAME=RNV_SERVER_DIR_NAME;

var ANDROID='android';exports.ANDROID=ANDROID;
var ANDROID_AUTO='androidauto';exports.ANDROID_AUTO=ANDROID_AUTO;
var ANDROID_TV='androidtv';exports.ANDROID_TV=ANDROID_TV;
var ANDROID_WEAR='androidwear';exports.ANDROID_WEAR=ANDROID_WEAR;
var ALEXA='alexa';exports.ALEXA=ALEXA;
var APPLE_AUTO='appleauto';exports.APPLE_AUTO=APPLE_AUTO;
var ASTIAN='astian';exports.ASTIAN=ASTIAN;
var BLACKBERRY='blackberry';exports.BLACKBERRY=BLACKBERRY;
var CHROMECAST='chromecast';exports.CHROMECAST=CHROMECAST;
var CHROME_OS='chromeos';exports.CHROME_OS=CHROME_OS;
var FIREFOX_OS='firefoxos';exports.FIREFOX_OS=FIREFOX_OS;
var FIREFOX_TV='firefoxtv';exports.FIREFOX_TV=FIREFOX_TV;
var FIRE_OS='fireos';exports.FIRE_OS=FIRE_OS;
var FIRE_TV='firetv';exports.FIRE_TV=FIRE_TV;
var HBBTV='hbbtv';exports.HBBTV=HBBTV;
var IOS='ios';exports.IOS=IOS;
var KAIOS='kaios';exports.KAIOS=KAIOS;
var MACOS='macos';exports.MACOS=MACOS;
var MEEGO='meego';exports.MEEGO=MEEGO;
var NETCAST='netcast';exports.NETCAST=NETCAST;
var OCCULUS='occulus';exports.OCCULUS=OCCULUS;
var ORSAY='orsay';exports.ORSAY=ORSAY;
var PS4='ps4';exports.PS4=PS4;
var ROKU='roku';exports.ROKU=ROKU;
var SAILFISH='sailfish';exports.SAILFISH=SAILFISH;
var TIVO='tivo';exports.TIVO=TIVO;
var TIZEN='tizen';exports.TIZEN=TIZEN;
var TIZEN_WATCH='tizenwatch';exports.TIZEN_WATCH=TIZEN_WATCH;
var TIZEN_MOBILE='tizenmobile';exports.TIZEN_MOBILE=TIZEN_MOBILE;
var TVOS='tvos';exports.TVOS=TVOS;
var UBUNTU='ubuntu';exports.UBUNTU=UBUNTU;
var UBUNTU_TOUCH='ubuntutouch';exports.UBUNTU_TOUCH=UBUNTU_TOUCH;
var UNITY='unity';exports.UNITY=UNITY;
var VEWD='vewd';exports.VEWD=VEWD;
var VIDAA='vidaa';exports.VIDAA=VIDAA;
var VIERACONNECT='vieraconnect';exports.VIERACONNECT=VIERACONNECT;
var VIZIO='vizio';exports.VIZIO=VIZIO;
var WATCHOS='watchos';exports.WATCHOS=WATCHOS;
var WEB='web';exports.WEB=WEB;
var WEBOS='webos';exports.WEBOS=WEBOS;
var WEBIAN='webian';exports.WEBIAN=WEBIAN;
var WII='wii';exports.WII=WII;
var WINDOWS='windows';exports.WINDOWS=WINDOWS;
var WP10='wp10';exports.WP10=WP10;
var WP8='wp8';exports.WP8=WP8;
var XBOX='xbox';exports.XBOX=XBOX;
var XBOX360='xbox360';exports.XBOX360=XBOX360;


var ICONS={
PHONE:'📱',
AUTO:'🚗',
TV:'📺',
CONSOLE:'🎮',
WATCH:'⌚',
DESKTOP:'🖥️ ',
BROWSER:'🌐',
VOICE:'📢',
SERVICE:'☁️'};exports.ICONS=ICONS;


var OS_WINDOWS='windows';exports.OS_WINDOWS=OS_WINDOWS;
var OS_MAC='mac';exports.OS_MAC=OS_MAC;
var OS_LINUX='linux';exports.OS_LINUX=OS_LINUX;

var OS={
ALL:[OS_MAC,OS_WINDOWS,OS_LINUX],
MAC_ONLY:[OS_MAC],
WINDOWS_ONLY:[OS_WINDOWS],
LINUX_ONLY:[OS_LINUX]};exports.OS=OS;


var EXT_BROWSER=['browser.jsx','browser.js','browser.tsx','browser.ts'];
var EXT_MOBILE=['mobile.jsx','mobile.js','mobile.tsx','mobile.ts'];
var EXT_MOBILE_NATIVE=['mobile.native.jsx','mobile.native.js','mobile.native.tsx','mobile.native.ts'];
var EXT_NATIVE=['native.jsx','native.js','native.tsx','native.ts'];
var EXT_TV_NATIVE=['tv.native.jsx','tv.native.js','tv.native.tsx','tv.native.ts'];
var EXT_IOS_MOBILE=['ios.mobile.jsx','ios.mobile.js','ios.mobile.tsx','ios.mobile.ts'];
var EXT_IOS=['ios.jsx','ios.js','ios.tsx','ios.ts'];
var EXT_ANDROID_MOBILE=['android.mobile.jsx','android.mobile.js','android.mobile.tsx','android.mobile.ts'];
var EXT_ANDROID=['android.jsx','android.js','android.tsx','android.ts'];
var EXT_TVOS=['tvos.jsx','tvos.js','tvos.tsx','tvos.ts'];
var EXT_TVOS_TV=['tvos.tv.jsx','tvos.tv.js','tvos.tv.tsx','tvos.tv.ts'];
var EXT_TV=['tv.jsx','tv.js','tv.tsx','tv.ts'];
var EXT_ANDROIDTV_TV=['androidtv.tv.jsx','androidtv.tv.js','androidtv.tv.tsx','androidtv.tv.ts'];
var EXT_ANDROIDTV=['androidtv.jsx','androidtv.js','androidtv.tsx','androidtv.ts'];
var EXT_SERVER_WEB=['server.web.jsx','server.web.js','server.web.tsx','server.web.ts'];
var EXT_WEB=['web.jsx','web.js','web.tsx','web.ts'];
var EXT_MACOS_DESKTOP=['macos.desktop.jsx','macos.desktop.js','macos.desktop.tsx','macos.desktop.ts'];
var EXT_DESKTOP=['desktop.jsx','desktop.js','desktop.tsx','desktop.ts'];
var EXT_MACOS=['macos.jsx','macos.js','macos.tsx','macos.ts'];
var EXT_DESKTOP_WEB=['desktop.web.jsx','desktop.web.js','desktop.web.tsx','desktop.web.ts'];
var EXT_ELECTRON=['electron.jsx','electron.js','electron.tsx','electron.ts'];
var EXT_TIZEN_TV=['tizen.tv.jsx','tizen.tv.js','tizen.tv.tsx','tizen.tv.ts'];
var EXT_TIZEN=['tizen.jsx','tizen.js','tizen.tsx','tizen.ts'];
var EXT_WEB_TV=['web.tv.jsx','web.tv.js','web.tv.tsx','web.tv.ts'];
var EXT_TV_WEB=['tv.web.jsx','tv.web.js','tv.web.tsx','tv.web.ts'];
var EXT_WEBOS_TV=['webos.tv.jsx','webos.tv.js','webos.tv.tsx','webos.tv.ts'];
var EXT_WEBOS=['webos.jsx','webos.js','webos.tsx','webos.ts'];
var EXT_ANDROIDWEAR_WATCH=['androidwear.watch.jsx','androidwear.watch.js','androidwear.watch.tsx','androidwear.watch.ts'];
var EXT_ANDROIDWEAR=['androidwear.jsx','androidwear.js','androidwear.tsx','androidwear.ts'];
var EXT_WATCH=['watch.jsx','watch.js','watch.tsx','watch.ts'];
var EXT_WATCH_NATIVE=['watch.native.jsx','watch.native.js','watch.native.tsx','watch.native.ts'];
var EXT_TIZENWATCH_WATCH=['tizenwatch.watch.jsx','tizenwatch.watch.js','tizenwatch.watch.tsx','tizenwatch.watch.ts'];
var EXT_TIZENWATCH=['tizenwatch.jsx','tizenwatch.js','tizenwatch.tsx','tizenwatch.ts'];
var EXT_WATCH_WEB=['watch.web.jsx','watch.web.js','watch.web.tsx','watch.web.ts'];
var EXT_TIZENMOBILE_MOBILE=['tizenmobile.mobile.jsx','tizenmobile.mobile.js','tizenmobile.mobile.tsx','tizenmobile.mobile.ts'];
var EXT_TIZENMOBILE=['tizenmobile.jsx','tizenmobile.js','tizenmobile.tsx','tizenmobile.ts'];
var EXT_MOBILE_WEB=['mobile.web.jsx','mobile.web.js','mobile.web.tsx','mobile.web.ts'];
var EXT_WIN_DESKTOP=['windows.desktop.jsx','windows.desktop.js','windows.desktop.tsx','windows.desktop.ts'];
var EXT_WIN=['windows.jsx','windows.js','windows.tsx','windows.ts'];
var EXT_KAIOS_MOBILE=['kaios.mobile.jsx','kaios.mobile.js','kaios.mobile.tsx','kaios.mobile.ts'];
var EXT_KAIOS=['kaios.jsx','kaios.js','kaios.tsx','kaios.ts'];
var EXT_FFOS_MOBILE=['firefoxos.mobile.jsx','firefoxos.mobile.js','firefoxos.mobile.tsx','firefoxos.mobile.ts'];
var EXT_FFOS=['firefoxos.jsx','firefoxos.js','firefoxos.tsx','firefoxos.ts'];
var EXT_FFTV_TV=['firefoxtv.tv.jsx','firefoxtv.tv.js','firefoxtv.tv.tsx','firefoxtv.tv.ts'];
var EXT_FFTV=['firefoxtv.jsx','firefoxtv.js','firefoxtv.tsx','firefoxtv.ts'];
var EXT_CAST_TV=['chromecast.tv.jsx','chromecast.tv.js','chromecast.tv.tsx','chromecast.tv.ts'];
var EXT_CAST=['chromecast.jsx','chromecast.js','chromecast.tsx','chromecast.ts'];
var EXT_FALLBACK_WEB=['mjs','jsx','js','json','wasm','tsx','ts'];
var EXT_FALLBACK=['jsx','js','json','wasm','tsx','ts'];


var REMOTE_DEBUG_PORT=8079;exports.REMOTE_DEBUG_PORT=REMOTE_DEBUG_PORT;

var EXT_LEGACY_NEXT=['next.jsx','next.js','next.tsx','next.ts'];
var EXT_LEGACY_SERVER_NEXT=['server.next.jsx','server.next.js','server.next.tsx','server.next.ts'];


var PLATFORMS={

web:{
defaultPort:8080,
icon:ICONS.BROWSER,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_BROWSER),
platforms:[].concat(EXT_LEGACY_SERVER_NEXT,EXT_SERVER_WEB,EXT_LEGACY_NEXT,EXT_WEB),
fallbacks:[].concat(EXT_FALLBACK_WEB)}},


ios:{
defaultPort:8082,
icon:ICONS.PHONE,
supportedOS:OS.MAC_ONLY,
isActive:true,
sourceExts:{
factors:[].concat(EXT_IOS_MOBILE,EXT_MOBILE),
platforms:[].concat(EXT_IOS),
fallbacks:[].concat(EXT_MOBILE_NATIVE,EXT_NATIVE,EXT_FALLBACK)}},


android:{
defaultPort:8083,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_ANDROID_MOBILE,EXT_MOBILE),
platforms:[].concat(EXT_ANDROID),
fallbacks:[].concat(EXT_MOBILE_NATIVE,EXT_NATIVE,EXT_FALLBACK)}},


androidtv:{
defaultPort:8084,
icon:ICONS.TV,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_ANDROIDTV_TV,EXT_TV),
platforms:[].concat(EXT_ANDROIDTV,EXT_ANDROID),
fallbacks:[].concat(EXT_TV_NATIVE,EXT_NATIVE,EXT_FALLBACK)}},


tvos:{
defaultPort:8085,
icon:ICONS.TV,
supportedOS:OS.MAC_ONLY,
isActive:true,
sourceExts:{
factors:[].concat(EXT_TVOS_TV,EXT_TV),
platforms:[].concat(EXT_TVOS,EXT_IOS),
fallbacks:[].concat(EXT_TV_NATIVE,EXT_NATIVE,EXT_FALLBACK)}},


macos:{
defaultPort:8086,
icon:ICONS.DESKTOP,
supportedOS:OS.MAC_ONLY,
isActive:true,
sourceExts:{
factors:[].concat(EXT_MACOS_DESKTOP,EXT_DESKTOP),
platforms:[].concat(EXT_MACOS),
fallbacks:[].concat(EXT_DESKTOP_WEB,EXT_ELECTRON,EXT_WEB,EXT_FALLBACK_WEB)}},


tizen:{
defaultPort:8087,
icon:ICONS.TV,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_TIZEN_TV,EXT_WEB_TV,EXT_TV),
platforms:[].concat(EXT_TIZEN),
fallbacks:[].concat(EXT_TV_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},


webos:{
defaultPort:8088,
icon:ICONS.TV,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_WEBOS_TV,EXT_WEB_TV,EXT_TV),
platforms:[].concat(EXT_WEBOS),
fallbacks:[].concat(EXT_TV_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},


androidwear:{
defaultPort:8089,
icon:ICONS.WATCH,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_ANDROIDWEAR_WATCH,EXT_WATCH),
platforms:[].concat(EXT_ANDROIDWEAR,EXT_ANDROID),
fallbacks:[].concat(EXT_WATCH_NATIVE,EXT_NATIVE,EXT_FALLBACK)}},


tizenwatch:{
defaultPort:8090,
icon:ICONS.WATCH,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_TIZENWATCH_WATCH,EXT_WATCH),
platforms:[].concat(EXT_TIZENWATCH),
fallbacks:[].concat(EXT_WATCH_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},


tizenmobile:{
defaultPort:8091,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_TIZENMOBILE_MOBILE,EXT_MOBILE),
platforms:[].concat(EXT_TIZENMOBILE),
fallbacks:[].concat(EXT_MOBILE_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},


windows:{
defaultPort:8092,
icon:ICONS.DESKTOP,
supportedOS:OS.WINDOWS_ONLY,
isActive:true,
sourceExts:{
factors:[].concat(EXT_WIN_DESKTOP,EXT_DESKTOP),
platforms:[].concat(EXT_WIN),
fallbacks:[].concat(EXT_DESKTOP_WEB,EXT_ELECTRON,EXT_WEB,EXT_FALLBACK_WEB)}},


kaios:{
defaultPort:8093,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_KAIOS_MOBILE,EXT_MOBILE),
platforms:[].concat(EXT_KAIOS),
fallbacks:[].concat(EXT_MOBILE_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},


firefoxos:{
defaultPort:8094,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_FFOS_MOBILE,EXT_MOBILE),
platforms:[].concat(EXT_FFOS),
fallbacks:[].concat(EXT_MOBILE_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},


firefoxtv:{
defaultPort:8014,
icon:ICONS.TV,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_FFTV_TV,EXT_WEB_TV,EXT_TV),
platforms:[].concat(EXT_FFTV),
fallbacks:[].concat(EXT_TV_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},


chromecast:{
defaultPort:8095,
icon:ICONS.TV,
supportedOS:OS.ALL,
isActive:true,
sourceExts:{
factors:[].concat(EXT_CAST_TV,EXT_WEB_TV,EXT_TV),
platforms:[].concat(EXT_CAST),
fallbacks:[].concat(EXT_TV_WEB,EXT_WEB,EXT_FALLBACK_WEB)}},



watchos:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

androidauto:{
defaultPort:8081,
icon:ICONS.AUTO,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

alexa:{
defaultPort:999999,
icon:ICONS.VOICE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

appleauto:{
defaultPort:8081,
icon:ICONS.AUTO,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

astian:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

blackberry:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

chromeos:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

fireos:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

firetv:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

hbbtv:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

meego:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

netcast:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

occulus:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

orsay:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

ps4:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

roku:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

sailfish:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

tivo:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

ubuntu:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

ubuntutouch:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

unity:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

vewd:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

vieraconnect:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

vizio:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

webian:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

wii:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

wp10:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

wp8:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

xbox:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]},

xbox360:{
defaultPort:999999,
icon:ICONS.PHONE,
supportedOS:OS.ALL,
isActive:false,
sourceExts:[]}};exports.PLATFORMS=PLATFORMS;



var EXTENSIONS={
web:[].concat((0,_toConsumableArray2.default)(PLATFORMS.web.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.web.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.web.sourceExts.fallbacks)),
ios:[].concat((0,_toConsumableArray2.default)(PLATFORMS.ios.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.ios.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.ios.sourceExts.fallbacks)),
android:[].concat((0,_toConsumableArray2.default)(PLATFORMS.android.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.android.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.android.sourceExts.fallbacks)),
androidtv:[].concat((0,_toConsumableArray2.default)(PLATFORMS.androidtv.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.androidtv.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.androidtv.sourceExts.fallbacks)),
tvos:[].concat((0,_toConsumableArray2.default)(PLATFORMS.tvos.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.tvos.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.tvos.sourceExts.fallbacks)),
macos:[].concat((0,_toConsumableArray2.default)(PLATFORMS.macos.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.macos.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.macos.sourceExts.fallbacks)),
tizen:[].concat((0,_toConsumableArray2.default)(PLATFORMS.tizen.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.tizen.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.tizen.sourceExts.fallbacks)),
webos:[].concat((0,_toConsumableArray2.default)(PLATFORMS.webos.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.webos.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.webos.sourceExts.fallbacks)),
androidwear:[].concat((0,_toConsumableArray2.default)(PLATFORMS.androidwear.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.androidwear.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.androidwear.sourceExts.fallbacks)),
tizenwatch:[].concat((0,_toConsumableArray2.default)(PLATFORMS.tizenwatch.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.tizenwatch.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.tizenwatch.sourceExts.fallbacks)),
tizenmobile:[].concat((0,_toConsumableArray2.default)(PLATFORMS.tizenmobile.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.tizenmobile.sourceExts.platforms),(0,_toConsumableArray2.default)(
PLATFORMS.tizenmobile.sourceExts.fallbacks)),
windows:[].concat((0,_toConsumableArray2.default)(PLATFORMS.windows.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.windows.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.windows.sourceExts.fallbacks)),
kaios:[].concat((0,_toConsumableArray2.default)(PLATFORMS.kaios.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.kaios.sourceExts.platforms),(0,_toConsumableArray2.default)(
PLATFORMS.kaios.sourceExts.fallbacks)),
firefoxos:[].concat((0,_toConsumableArray2.default)(PLATFORMS.firefoxos.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.firefoxos.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.firefoxos.sourceExts.fallbacks)),
firefoxtv:[].concat((0,_toConsumableArray2.default)(PLATFORMS.firefoxtv.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.firefoxtv.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.firefoxtv.sourceExts.fallbacks)),
chromecast:[].concat((0,_toConsumableArray2.default)(PLATFORMS.chromecast.sourceExts.factors),(0,_toConsumableArray2.default)(
PLATFORMS.chromecast.sourceExts.platforms),(0,_toConsumableArray2.default)(PLATFORMS.chromecast.sourceExts.fallbacks))};exports.EXTENSIONS=EXTENSIONS;


var WEB_HOSTED_PLATFORMS=[
WEB,
TIZEN,
WEBOS,
MACOS,
WINDOWS,
TIZEN_MOBILE,
TIZEN_WATCH];exports.WEB_HOSTED_PLATFORMS=WEB_HOSTED_PLATFORMS;


var PLATFORM_GROUP_SMARTTV='smarttv';exports.PLATFORM_GROUP_SMARTTV=PLATFORM_GROUP_SMARTTV;
var PLATFORM_GROUP_ELECTRON='electron';exports.PLATFORM_GROUP_ELECTRON=PLATFORM_GROUP_ELECTRON;

var FORM_FACTOR_MOBILE='mobile';exports.FORM_FACTOR_MOBILE=FORM_FACTOR_MOBILE;
var FORM_FACTOR_DESKTOP='desktop';exports.FORM_FACTOR_DESKTOP=FORM_FACTOR_DESKTOP;
var FORM_FACTOR_WATCH='watch';exports.FORM_FACTOR_WATCH=FORM_FACTOR_WATCH;
var FORM_FACTOR_TV='tv';exports.FORM_FACTOR_TV=FORM_FACTOR_TV;

var CLI_ANDROID_EMULATOR='androidEmulator';exports.CLI_ANDROID_EMULATOR=CLI_ANDROID_EMULATOR;
var CLI_ANDROID_ADB='androidAdb';exports.CLI_ANDROID_ADB=CLI_ANDROID_ADB;
var CLI_ANDROID_AVDMANAGER='androidAvdManager';exports.CLI_ANDROID_AVDMANAGER=CLI_ANDROID_AVDMANAGER;
var CLI_ANDROID_SDKMANAGER='androidSdkManager';exports.CLI_ANDROID_SDKMANAGER=CLI_ANDROID_SDKMANAGER;
var CLI_TIZEN_EMULATOR='tizenEmulator';exports.CLI_TIZEN_EMULATOR=CLI_TIZEN_EMULATOR;
var CLI_KAIOS_EMULATOR='tizenEmulator';exports.CLI_KAIOS_EMULATOR=CLI_KAIOS_EMULATOR;
var CLI_TIZEN='tizen';exports.CLI_TIZEN=CLI_TIZEN;
var CLI_SDB_TIZEN='tizenSdb';exports.CLI_SDB_TIZEN=CLI_SDB_TIZEN;
var CLI_WEBOS_ARES='webosAres';exports.CLI_WEBOS_ARES=CLI_WEBOS_ARES;
var CLI_WEBOS_ARES_PACKAGE='webosAresPackage';exports.CLI_WEBOS_ARES_PACKAGE=CLI_WEBOS_ARES_PACKAGE;
var CLI_WEBOS_ARES_INSTALL='webosAresInstall';exports.CLI_WEBOS_ARES_INSTALL=CLI_WEBOS_ARES_INSTALL;
var CLI_WEBOS_ARES_LAUNCH='webosAresLaunch';exports.CLI_WEBOS_ARES_LAUNCH=CLI_WEBOS_ARES_LAUNCH;
var CLI_WEBOS_ARES_SETUP_DEVICE='webosAresSetup';exports.CLI_WEBOS_ARES_SETUP_DEVICE=CLI_WEBOS_ARES_SETUP_DEVICE;
var CLI_WEBOS_ARES_DEVICE_INFO='webosAresDeviceInfo';exports.CLI_WEBOS_ARES_DEVICE_INFO=CLI_WEBOS_ARES_DEVICE_INFO;
var CLI_WEBOS_ARES_NOVACOM='webosAresNovacom';exports.CLI_WEBOS_ARES_NOVACOM=CLI_WEBOS_ARES_NOVACOM;

var ANDROID_SDK='ANDROID_SDK';exports.ANDROID_SDK=ANDROID_SDK;
var ANDROID_NDK='ANDROID_NDK';exports.ANDROID_NDK=ANDROID_NDK;
var TIZEN_SDK='TIZEN_SDK';exports.TIZEN_SDK=TIZEN_SDK;
var WEBOS_SDK='WEBOS_SDK';exports.WEBOS_SDK=WEBOS_SDK;
var KAIOS_SDK='KAIOS_SDK';exports.KAIOS_SDK=KAIOS_SDK;

var RENATIVE_CONFIG_NAME='renative.json';exports.RENATIVE_CONFIG_NAME=RENATIVE_CONFIG_NAME;
var RENATIVE_CONFIG_LOCAL_NAME='renative.local.json';exports.RENATIVE_CONFIG_LOCAL_NAME=RENATIVE_CONFIG_LOCAL_NAME;
var RENATIVE_CONFIG_PRIVATE_NAME='renative.private.json';exports.RENATIVE_CONFIG_PRIVATE_NAME=RENATIVE_CONFIG_PRIVATE_NAME;
var RENATIVE_CONFIG_TEMPLATE_NAME='renative.template.json';exports.RENATIVE_CONFIG_TEMPLATE_NAME=RENATIVE_CONFIG_TEMPLATE_NAME;
var RENATIVE_CONFIG_BUILD_NAME='renative.build.json';exports.RENATIVE_CONFIG_BUILD_NAME=RENATIVE_CONFIG_BUILD_NAME;
var RENATIVE_CONFIG_RUNTIME_NAME='renative.runtime.json';exports.RENATIVE_CONFIG_RUNTIME_NAME=RENATIVE_CONFIG_RUNTIME_NAME;
var RENATIVE_CONFIG_WORKSPACES_NAME='renative.workspaces.json';exports.RENATIVE_CONFIG_WORKSPACES_NAME=RENATIVE_CONFIG_WORKSPACES_NAME;
var RENATIVE_CONFIG_PLUGINS_NAME='renative.plugins.json';exports.RENATIVE_CONFIG_PLUGINS_NAME=RENATIVE_CONFIG_PLUGINS_NAME;
var RENATIVE_CONFIG_TEMPLATES_NAME='renative.templates.json';exports.RENATIVE_CONFIG_TEMPLATES_NAME=RENATIVE_CONFIG_TEMPLATES_NAME;
var RENATIVE_CONFIG_PLATFORMS_NAME='renative.platforms.json';exports.RENATIVE_CONFIG_PLATFORMS_NAME=RENATIVE_CONFIG_PLATFORMS_NAME;
var RENATIVE_CONFIG_ENGINES_NAME='renative.engines.json';exports.RENATIVE_CONFIG_ENGINES_NAME=RENATIVE_CONFIG_ENGINES_NAME;
var RN_CLI_CONFIG_NAME='metro.config.js';exports.RN_CLI_CONFIG_NAME=RN_CLI_CONFIG_NAME;
var RN_BABEL_CONFIG_NAME='babel.config.js';exports.RN_BABEL_CONFIG_NAME=RN_BABEL_CONFIG_NAME;
var NEXT_CONFIG_NAME='next.config.js';exports.NEXT_CONFIG_NAME=NEXT_CONFIG_NAME;
var SAMPLE_APP_ID='helloworld';exports.SAMPLE_APP_ID=SAMPLE_APP_ID;

var IS_TABLET_ABOVE_INCH=6.5;exports.IS_TABLET_ABOVE_INCH=IS_TABLET_ABOVE_INCH;
var IS_WEAR_UNDER_SIZE=1000;exports.IS_WEAR_UNDER_SIZE=IS_WEAR_UNDER_SIZE;

var PACKAGE_JSON_FILEDS=[
'name',
'version',
'description',
'keywords',
'homepage',
'bugs',
'license',
'author',
'contributors',
'files',
'main',
'browser',
'bin',
'man',
'directories',
'repository',
'scripts',
'config',
'dependencies',
'devDependencies',
'peerDependencies',
'bundledDependencies',
'optionalDependencies',
'engines',
'engineStrict',
'os',
'cpu',
'private',
'publishConfig'];exports.PACKAGE_JSON_FILEDS=PACKAGE_JSON_FILEDS;

var SUPPORTED_PLATFORMS=[
IOS,
ANDROID,
ANDROID_TV,
ANDROID_WEAR,
WEB,
TIZEN,
TIZEN_MOBILE,
TVOS,
WEBOS,
MACOS,
WINDOWS,
TIZEN_WATCH,
KAIOS,
FIREFOX_OS,
FIREFOX_TV,
CHROMECAST];exports.SUPPORTED_PLATFORMS=SUPPORTED_PLATFORMS;

var SUPPORTED_PLATFORMS_MAC=[
IOS,
ANDROID,
ANDROID_TV,
ANDROID_WEAR,
WEB,
TIZEN_MOBILE,
TIZEN,
TVOS,
WEBOS,
MACOS,
WINDOWS,
TIZEN_WATCH,
KAIOS,
FIREFOX_OS,
FIREFOX_TV];exports.SUPPORTED_PLATFORMS_MAC=SUPPORTED_PLATFORMS_MAC;

var SUPPORTED_PLATFORMS_WIN=[
ANDROID,
ANDROID_TV,
ANDROID_WEAR,
WEB,
TIZEN,
TVOS,
TIZEN_MOBILE,
WEBOS,
WINDOWS,
TIZEN_WATCH,
KAIOS,
FIREFOX_OS,
FIREFOX_TV];exports.SUPPORTED_PLATFORMS_WIN=SUPPORTED_PLATFORMS_WIN;

var SUPPORTED_PLATFORMS_LINUX=[ANDROID,ANDROID_TV,ANDROID_WEAR];exports.SUPPORTED_PLATFORMS_LINUX=SUPPORTED_PLATFORMS_LINUX;
var SDK_PLATFORMS={};exports.SDK_PLATFORMS=SDK_PLATFORMS;
SDK_PLATFORMS[ANDROID]=ANDROID_SDK;
SDK_PLATFORMS[ANDROID_TV]=ANDROID_SDK;
SDK_PLATFORMS[ANDROID_WEAR]=ANDROID_SDK;
SDK_PLATFORMS[TIZEN]=TIZEN_SDK;
SDK_PLATFORMS[TIZEN_WATCH]=TIZEN_SDK;
SDK_PLATFORMS[TIZEN_MOBILE]=TIZEN_SDK;
SDK_PLATFORMS[WEBOS]=WEBOS_SDK;
SDK_PLATFORMS[KAIOS]=KAIOS_SDK;

var TASK_RUN='run';exports.TASK_RUN=TASK_RUN;
var TASK_CONFIGURE='configure';exports.TASK_CONFIGURE=TASK_CONFIGURE;
var TASK_DOCTOR='doctor';exports.TASK_DOCTOR=TASK_DOCTOR;
var TASK_BUILD='build';exports.TASK_BUILD=TASK_BUILD;
var TASK_INFO='info';exports.TASK_INFO=TASK_INFO;
var TASK_START='start';exports.TASK_START=TASK_START;
var TASK_EXPORT='export';exports.TASK_EXPORT=TASK_EXPORT;
var TASK_DEBUG='debug';exports.TASK_DEBUG=TASK_DEBUG;
var TASK_PACKAGE='package';exports.TASK_PACKAGE=TASK_PACKAGE;
var TASK_DEPLOY='deploy';exports.TASK_DEPLOY=TASK_DEPLOY;
var TASK_LOG='log';exports.TASK_LOG=TASK_LOG;
var TASK_CLEAN='clean';exports.TASK_CLEAN=TASK_CLEAN;
var TASK_INSTALL='install';exports.TASK_INSTALL=TASK_INSTALL;
var TASK_PUBLISH='publish';exports.TASK_PUBLISH=TASK_PUBLISH;
var TASK_STATUS='status';exports.TASK_STATUS=TASK_STATUS;
var TASK_SWITCH='switch';exports.TASK_SWITCH=TASK_SWITCH;
var TASK_TARGET_LAUNCH='target launch';exports.TASK_TARGET_LAUNCH=TASK_TARGET_LAUNCH;
var TASK_TARGET_LIST='target list';exports.TASK_TARGET_LIST=TASK_TARGET_LIST;
var TASK_TARGET='target';exports.TASK_TARGET=TASK_TARGET;
var TASK_TEMPLATE_ADD='template add';exports.TASK_TEMPLATE_ADD=TASK_TEMPLATE_ADD;
var TASK_TEMPLATE_LIST='template list';exports.TASK_TEMPLATE_LIST=TASK_TEMPLATE_LIST;
var TASK_TEMPLATE_APPLY='template apply';exports.TASK_TEMPLATE_APPLY=TASK_TEMPLATE_APPLY;
var TASK_WORKSPACE_ADD='workspace add';exports.TASK_WORKSPACE_ADD=TASK_WORKSPACE_ADD;
var TASK_WORKSPACE_CONNECT='workspace connect';exports.TASK_WORKSPACE_CONNECT=TASK_WORKSPACE_CONNECT;
var TASK_WORKSPACE_LIST='workspace list';exports.TASK_WORKSPACE_LIST=TASK_WORKSPACE_LIST;
var TASK_WORKSPACE_UPDATE='workspace update';exports.TASK_WORKSPACE_UPDATE=TASK_WORKSPACE_UPDATE;
var TASK_PLATFORM_CONFIGURE='platform configure';exports.TASK_PLATFORM_CONFIGURE=TASK_PLATFORM_CONFIGURE;
var TASK_PLATFORM_CONNECT='platform connect';exports.TASK_PLATFORM_CONNECT=TASK_PLATFORM_CONNECT;
var TASK_PLATFORM_EJECT='platform eject';exports.TASK_PLATFORM_EJECT=TASK_PLATFORM_EJECT;
var TASK_PLATFORM_LIST='platform list';exports.TASK_PLATFORM_LIST=TASK_PLATFORM_LIST;
var TASK_PLATFORM_SETUP='platform setup';exports.TASK_PLATFORM_SETUP=TASK_PLATFORM_SETUP;
var TASK_PROJECT_CONFIGURE='project configure';exports.TASK_PROJECT_CONFIGURE=TASK_PROJECT_CONFIGURE;
var TASK_PROJECT_UPGRADE='project upgrade';exports.TASK_PROJECT_UPGRADE=TASK_PROJECT_UPGRADE;
var TASK_PLUGIN_ADD='plugin add';exports.TASK_PLUGIN_ADD=TASK_PLUGIN_ADD;
var TASK_PLUGIN_LIST='plugin list';exports.TASK_PLUGIN_LIST=TASK_PLUGIN_LIST;
var TASK_PLUGIN_UPDATE='plugin update';exports.TASK_PLUGIN_UPDATE=TASK_PLUGIN_UPDATE;
var TASK_CRYPTO_ENCRYPT='crypto encrypt';exports.TASK_CRYPTO_ENCRYPT=TASK_CRYPTO_ENCRYPT;
var TASK_CRYPTO_DECRYPT='crypto decrypt';exports.TASK_CRYPTO_DECRYPT=TASK_CRYPTO_DECRYPT;
var TASK_CRYPTO_INSTALL_CERTS='crypto installCerts';exports.TASK_CRYPTO_INSTALL_CERTS=TASK_CRYPTO_INSTALL_CERTS;
var TASK_CRYPTO_INSTALL_PROFILES='crypto installProfiles';exports.TASK_CRYPTO_INSTALL_PROFILES=TASK_CRYPTO_INSTALL_PROFILES;
var TASK_CRYPTO_INSTALL_PROFILE='crypto installProfile';exports.TASK_CRYPTO_INSTALL_PROFILE=TASK_CRYPTO_INSTALL_PROFILE;
var TASK_CRYPTO_UPDATE_PROFILE='crypto updateProfile';exports.TASK_CRYPTO_UPDATE_PROFILE=TASK_CRYPTO_UPDATE_PROFILE;
var TASK_CRYPTO_UPDATE_PROFILES='crypto updateProfiles';exports.TASK_CRYPTO_UPDATE_PROFILES=TASK_CRYPTO_UPDATE_PROFILES;
var TASK_HOOKS_RUN='hooks run';exports.TASK_HOOKS_RUN=TASK_HOOKS_RUN;
var TASK_HOOKS_LIST='hooks list';exports.TASK_HOOKS_LIST=TASK_HOOKS_LIST;
var TASK_HOOKS_PIPES='hooks pipes';exports.TASK_HOOKS_PIPES=TASK_HOOKS_PIPES;
var TASK_PKG='pkg';exports.TASK_PKG=TASK_PKG;
var TASK_APP_CONFIGURE='app configure';exports.TASK_APP_CONFIGURE=TASK_APP_CONFIGURE;
var TASK_APP_CREATE='app create';exports.TASK_APP_CREATE=TASK_APP_CREATE;
var TASK_WORKSPACE_CONFIGURE='workspace configure';exports.TASK_WORKSPACE_CONFIGURE=TASK_WORKSPACE_CONFIGURE;
var TASK_CONFIGURE_SOFT='configureSoft';exports.TASK_CONFIGURE_SOFT=TASK_CONFIGURE_SOFT;
var TASK_KILL='kill';exports.TASK_KILL=TASK_KILL;

var _PARAMS={
info:{
shortcut:'i',
value:'value',
description:'Show full debug Info'},

updatePods:{
shortcut:'u',
description:'Force update dependencies (iOS only)'},

platform:{
shortcut:'p',
value:'value',
description:'select specific Platform'},

appConfigID:{
shortcut:'c',
value:'value',
description:'select specific app Config id'},

target:{
shortcut:'t',
value:'value',
description:'select specific Target device/simulator'},

template:{
shortcut:'T',
value:'value',
isRequired:true,
description:'select specific template'},

device:{
shortcut:'d',
value:'value',
description:'select connected Device'},

scheme:{
shortcut:'s',
value:'value',
description:'select build Scheme'},

filter:{
shortcut:'f',
value:'value',
isRequired:true,
description:'Filter value'},

list:{
shortcut:'l',
description:'return List of items related to command'},

only:{
shortcut:'o',
description:'run Only top command (Skip dependencies)'},

reset:{
shortcut:'r',
description:'also perform Reset of platform'},

resetHard:{
shortcut:'R',
description:'also perform Reset of platform and all assets'},

key:{
shortcut:'k',
value:'value',
isRequired:true,
description:'Pass the key/password'},

blueprint:{
shortcut:'b',
value:'value',
description:'Blueprint for targets'},

help:{
shortcut:'h',
description:'Displays help info for particular command'},

host:{
shortcut:'H',
value:'value',
isRequired:true,
description:'custom Host ip'},

exeMethod:{
shortcut:'x',
value:'value',
isRequired:true,
description:'eXecutable method in buildHooks'},

port:{
shortcut:'P',
value:'value',
isRequired:true,
description:'custom Port'},

debug:{
shortcut:'D',
value:'value',
description:'enable or disable remote debugger.',
examples:[
'--debug weinre //run remote debug with weinre as preference',
'--debug chii //run remote debug with chii as preference',
'--debug false //force disable remote debug',
'--debug //run remote debug with default preference (chii)']},


global:{
shortcut:'G',
description:'Flag for setting a config value for all RNV projects'},

engine:{
shortcut:'e',
value:'value',
isRequired:true,
description:'engine to be used (next)'},

debugIp:{
value:'value',
isRequired:true,
description:'(optional) overwrite the ip to which the remote debugger will connect'},

ci:{
description:'CI/CD flag so it wont ask questions'},

mono:{
description:'Monochrome console output without chalk'},

skipNotifications:{
description:'Skip sending any integrated notifications'},

keychain:{
value:'value',
isRequired:true,
description:'Name of the keychain'},

provisioningStyle:{
value:'value',
isRequired:true,
description:'Set provisioningStyle <Automatic | Manual>'},

codeSignIdentity:{
value:'value',
isRequired:true,
description:'Set codeSignIdentity ie <iPhone Distribution>'},

provisionProfileSpecifier:{
value:'value',
isRequired:true,
description:'Name of provisionProfile'},

hosted:{
description:'Run in a hosted environment (skip budleAssets)'},

maxErrorLength:{
value:'number',
isRequired:true,
description:'Specify how many characters each error should display. Default 200'},

skipTargetCheck:{
description:'Skip Android target check, just display the raw adb devices to choose from'},

analyzer:{
description:'Enable real-time bundle analyzer'},

xcodebuildArchiveArgs:{
value:'value',
isRequired:true,
description:'pass down custom xcodebuild arguments'},

xcodebuildExportArgs:{
value:'value',
isRequired:true,
description:'pass down custom xcodebuild arguments'}};



Object.keys(_PARAMS).forEach(function(k){
_PARAMS[k].key=k;
});


var PARAMS={
withBase:function withBase(arr){return[_PARAMS.info,_PARAMS.ci,_PARAMS.mono,_PARAMS.maxErrorLength,_PARAMS.only].concat(arr||[]);},
withConfigure:function withConfigure(arr){return[_PARAMS.reset,_PARAMS.resetHard,_PARAMS.engine,
_PARAMS.appConfigID,_PARAMS.scheme,_PARAMS.platform].concat(arr||[]);},
withRun:function withRun(arr){return[_PARAMS.target,_PARAMS.device,_PARAMS.hosted,
_PARAMS.port,_PARAMS.debug,_PARAMS.debugIp,_PARAMS.skipTargetCheck,_PARAMS.host].concat(arr||[]);},
withAll:function withAll(arr){return Object.values(_PARAMS).concat(arr||[]);},
all:Object.keys(_PARAMS)};exports.PARAMS=PARAMS;



var configSchema={
analytics:{
values:['true','false'],
key:'enableAnalytics',
default:true}};exports.configSchema=configSchema;




var INJECTABLE_CONFIG_PROPS=['id','title','entryFile','backgroundColor','scheme',
'teamID','provisioningStyle','bundleAssets','enableHermes','universalApk','multipleAPKs','pagesDir'];exports.INJECTABLE_CONFIG_PROPS=INJECTABLE_CONFIG_PROPS;
var INJECTABLE_RUNTIME_PROPS=['appId','scheme','timestamp','localhost','target','port'];exports.INJECTABLE_RUNTIME_PROPS=INJECTABLE_RUNTIME_PROPS;

var REDASH_URL='https://rnv.nxg.staging.24imedia.com/events';exports.REDASH_URL=REDASH_URL;
var REDASH_KEY='zCYINQqMxvat1V41Hb9d69JMVBDNLyeQ4wICtdtD';exports.REDASH_KEY=REDASH_KEY;

var REMOTE_DEBUGGER_ENABLED_PLATFORMS=[TIZEN,TIZEN_MOBILE,TIZEN_WATCH];exports.REMOTE_DEBUGGER_ENABLED_PLATFORMS=REMOTE_DEBUGGER_ENABLED_PLATFORMS;
