// rnv/dist/platformTools/chromecast/index.js
var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.runChromecast=exports.configureProject=exports.configureChromecastProject=void 0;var _regenerator=_interopRequireDefault(require("@babel/runtime/regenerator"));
var _web=require("../web");
var _logger=require("../../systemTools/logger");
var _projectParser=require("../../projectTools/projectParser");

var configureChromecastProject=function configureChromecastProject(c){return _regenerator.default.async(function configureChromecastProject$(_context){while(1){switch(_context.prev=_context.next){case 0:
(0,_logger.logTask)("configureChromecastProject:"+c.platform);_context.next=3;return _regenerator.default.awrap(

(0,_projectParser.copyAssetsFolder)(c,c.platform));case 3:_context.next=5;return _regenerator.default.awrap(
(0,_web.configureCoreWebProject)(c,c.platform));case 5:_context.next=7;return _regenerator.default.awrap(
configureProject(c,c.platform));case 7:return _context.abrupt("return",
(0,_projectParser.copyBuildsFolder)(c,c.platform));case 8:case"end":return _context.stop();}}},null,null,null,Promise);};exports.configureChromecastProject=configureChromecastProject;


var configureProject=function configureProject(c){return _regenerator.default.async(function configureProject$(_context2){while(1){switch(_context2.prev=_context2.next){case 0:
(0,_logger.logTask)("configureProject:"+c.platform);case 1:case"end":return _context2.stop();}}},null,null,null,Promise);};exports.configureProject=configureProject;


var runChromecast=function runChromecast(c){return _regenerator.default.async(function runChromecast$(_context3){while(1){switch(_context3.prev=_context3.next){case 0:
(0,_logger.logTask)("runChromecast:"+c.platform);_context3.next=3;return _regenerator.default.awrap(
(0,_web.runWeb)(c,c.platform,c.runtime.port));case 3:case"end":return _context3.stop();}}},null,null,null,Promise);};exports.runChromecast=runChromecast;
