// rnv/dist/engine-rn-electron/commonEngine.js
var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.getPlatformProjectDir=exports.ejectPlatform=exports.getTemplateRootDir=exports.getTemplateProjectDir=exports.getPlatformOutputDir=exports.getPlatformBuildDir=void 0;var _path=_interopRequireDefault(require("path"));
var _common=require("../core/common");
var _constants=require("../core/constants");

var getPlatformBuildDir=function getPlatformBuildDir(c){return(0,_common.getAppFolder)(c);};exports.getPlatformBuildDir=getPlatformBuildDir;

var getPlatformOutputDir=function getPlatformOutputDir(c){
var dir=getPlatformBuildDir(c);
return dir;
};exports.getPlatformOutputDir=getPlatformOutputDir;

var getTemplateProjectDir=function getTemplateProjectDir(c){
var dir=(0,_common.getTemplateDir)(c);
var output;
switch(c.platform){
case _constants.MACOS:
case _constants.WINDOWS:
output=_path.default.join(dir,_constants.RNV_PROJECT_DIR_NAME);
break;
default:
output=dir;}

return output;
};exports.getTemplateProjectDir=getTemplateProjectDir;

var getTemplateRootDir=function getTemplateRootDir(c,platform){
var dir=c.paths.project.platformTemplatesDirs[platform];
return dir;
};exports.getTemplateRootDir=getTemplateRootDir;

var ejectPlatform=function ejectPlatform(){

};exports.ejectPlatform=ejectPlatform;

var getPlatformProjectDir=function getPlatformProjectDir(c){
var dir=getPlatformBuildDir(c);
var output;
switch(c.platform){
case _constants.MACOS:
case _constants.WINDOWS:
output=_path.default.join(dir,_constants.RNV_PROJECT_DIR_NAME);
break;
default:
output=dir;}

return output;
};exports.getPlatformProjectDir=getPlatformProjectDir;
