// rnv/dist/index.js
var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});Object.defineProperty(exports,"Config",{enumerable:true,get:function get(){return _config.default;}});Object.defineProperty(exports,"CLI",{enumerable:true,get:function get(){return _cli.default;}});exports.SchemaParser=exports.EngineManager=exports.Resolver=exports.SetupTools=exports.PluginTools=exports.Doctor=exports.FileUtils=exports.Exec=exports.Constants=exports.Prompt=exports.Logger=exports.Common=exports.default=exports.doResolvePath=exports.doResolve=exports.run=exports.initializeBuilder=void 0;var _regenerator=_interopRequireDefault(require("@babel/runtime/regenerator"));var Common=_interopRequireWildcard(require("./core/common"));exports.Common=Common;
var CoreUtils=_interopRequireWildcard(require("./core/utils"));
var Logger=_interopRequireWildcard(require("./core/systemManager/logger"));exports.Logger=Logger;
var Prompt=_interopRequireWildcard(require("./cli/prompt"));exports.Prompt=Prompt;
var ConfigParser=_interopRequireWildcard(require("./core/configManager/configParser"));
var Constants=_interopRequireWildcard(require("./core/constants"));exports.Constants=Constants;
var Exec=_interopRequireWildcard(require("./core/systemManager/exec"));exports.Exec=Exec;
var FileUtils=_interopRequireWildcard(require("./core/systemManager/fileutils"));exports.FileUtils=FileUtils;
var Doctor=_interopRequireWildcard(require("./core/systemManager/doctor"));exports.Doctor=Doctor;
var PluginTools=_interopRequireWildcard(require("./core/pluginManager"));exports.PluginTools=PluginTools;
var SetupTools=_interopRequireWildcard(require("./core/setupManager"));exports.SetupTools=SetupTools;
var Resolver=_interopRequireWildcard(require("./core/resolve"));exports.Resolver=Resolver;
var EngineManager=_interopRequireWildcard(require("./core/engineManager"));exports.EngineManager=EngineManager;
var SchemaParser=_interopRequireWildcard(require("./core/schemaManager/schemaParser"));exports.SchemaParser=SchemaParser;
var _analytics=_interopRequireDefault(require("./core/systemManager/analytics"));
var _config=_interopRequireDefault(require("./core/configManager/config"));
var _cli=_interopRequireDefault(require("./cli"));

require("source-map-support/register");

_analytics.default.initialize();

var initializeBuilder=function initializeBuilder(cmd,subCmd,process,program){var c;return _regenerator.default.async(function initializeBuilder$(_context){while(1){switch(_context.prev=_context.next){case 0:
FileUtils.configureFilesystem(Resolver.getConfigProp,Resolver.doResolve,CoreUtils.isSystemWin);
c=ConfigParser.createRnvConfig(program,process,cmd,subCmd);

Logger.configureLogger(c,_analytics.default);
Logger.logInitialize();return _context.abrupt("return",

c);case 5:case"end":return _context.stop();}}},null,null,null,Promise);};exports.initializeBuilder=initializeBuilder;


var run=function run(cmd,subCmd,program,process){
initializeBuilder(cmd,subCmd,process,program).
then(function(c){return _config.default.initializeConfig(c);}).
then(function(c){return(0,_cli.default)(c);}).
then(function(){return Logger.logComplete(!_config.default.getConfig().runtime.keepSessionActive);}).
catch(function(e){return Logger.logError(e,true);});
};exports.run=run;var




















doResolve=Resolver.doResolve;exports.doResolve=doResolve;var
doResolvePath=Resolver.doResolvePath;exports.doResolvePath=doResolvePath;var _default=

{run:run,Config:_config.default};exports.default=_default;
