// rnv/dist/sdk-webpack/chromecast.js
var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.runChromecast=exports.configureProject=exports.configureChromecastProject=void 0;var _regenerator=_interopRequireDefault(require("@babel/runtime/regenerator"));var _index=require("./index");
var _logger=require("../core/systemManager/logger");
var _projectParser=require("../core/projectManager/projectParser");
var _common=require("../core/common");
var _constants=require("../core/constants");

var configureChromecastProject=function configureChromecastProject(c){var bundleAssets;return _regenerator.default.async(function configureChromecastProject$(_context){while(1){switch(_context.prev=_context.next){case 0:
(0,_logger.logTask)("configureChromecastProject:"+c.platform);

c.runtime.platformBuildsProjectPath=""+(0,_common.getPlatformProjectDir)(c);

bundleAssets=(0,_common.getConfigProp)(c,c.platform,'bundleAssets');_context.next=5;return _regenerator.default.awrap(

(0,_projectParser.copyAssetsFolder)(c,c.platform));case 5:_context.next=7;return _regenerator.default.awrap(
(0,_index.configureCoreWebProject)(c,bundleAssets?_constants.RNV_PROJECT_DIR_NAME:_constants.RNV_SERVER_DIR_NAME));case 7:_context.next=9;return _regenerator.default.awrap(
configureProject(c));case 9:return _context.abrupt("return",
(0,_projectParser.copyBuildsFolder)(c,c.platform));case 10:case"end":return _context.stop();}}},null,null,null,Promise);};exports.configureChromecastProject=configureChromecastProject;


var configureProject=function configureProject(c){return _regenerator.default.async(function configureProject$(_context2){while(1){switch(_context2.prev=_context2.next){case 0:
(0,_logger.logTask)("configureProject:"+c.platform);case 1:case"end":return _context2.stop();}}},null,null,null,Promise);};exports.configureProject=configureProject;


var runChromecast=function runChromecast(c){return _regenerator.default.async(function runChromecast$(_context3){while(1){switch(_context3.prev=_context3.next){case 0:
(0,_logger.logTask)("runChromecast:"+c.platform);_context3.next=3;return _regenerator.default.awrap(
(0,_index.runWebpackServer)(c));case 3:case"end":return _context3.stop();}}},null,null,null,Promise);};exports.runChromecast=runChromecast;
