// rnv/dist/engine-rn-web/index.js
var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _engineManager=require("../core/engineManager");
var _commonEngine=require("./commonEngine");


var _taskRnv=_interopRequireDefault(require("./tasks/task.rnv.run"));
var _taskRnv2=_interopRequireDefault(require("./tasks/task.rnv.package"));
var _taskRnv3=_interopRequireDefault(require("./tasks/task.rnv.build"));
var _taskRnv4=_interopRequireDefault(require("./tasks/task.rnv.configure"));
var _taskRnv5=_interopRequireDefault(require("./tasks/task.rnv.start"));
var _taskRnv6=_interopRequireDefault(require("./tasks/task.rnv.export"));
var _taskRnv7=_interopRequireDefault(require("./tasks/task.rnv.deploy"));
var _taskRnv8=_interopRequireDefault(require("./tasks/task.rnv.debug"));

var TASKS={};

var addTask=function addTask(taskInstance){
TASKS[taskInstance.task]=taskInstance;
};

addTask(_taskRnv.default);
addTask(_taskRnv2.default);
addTask(_taskRnv3.default);
addTask(_taskRnv4.default);
addTask(_taskRnv5.default);
addTask(_taskRnv6.default);
addTask(_taskRnv7.default);
addTask(_taskRnv8.default);

var executeTask=function executeTask(c,task,parentTask,originTask,isFirstTask){return(0,_engineManager.executeEngineTask)(
c,task,parentTask,originTask,TASKS,isFirstTask);};


var hasTask=function hasTask(task,isProjectScope){return(0,_engineManager.hasEngineTask)(task,TASKS,isProjectScope);};

var getTask=function getTask(task){return(0,_engineManager.getEngineTask)(task,TASKS);};

var getSubTasks=function getSubTasks(task,exactMatch){return(0,_engineManager.getEngineSubTasks)(task,TASKS,exactMatch);};

var getTasks=function getTasks(){return Object.values(TASKS);};

var getId=function getId(){return'engine-rn-web';};

var title='Engine RN Web';var _default=

{
getPlatformBuildDir:_commonEngine.getPlatformBuildDir,
getPlatformProjectDir:_commonEngine.getPlatformProjectDir,
getPlatformOutputDir:_commonEngine.getPlatformOutputDir,
ejectPlatform:_commonEngine.ejectPlatform,
getTemplateProjectDir:_commonEngine.getTemplateProjectDir,
getTemplateRootDir:_commonEngine.getTemplateRootDir,
executeTask:executeTask,
addTask:addTask,
hasTask:hasTask,
getTask:getTask,
getSubTasks:getSubTasks,
getTasks:getTasks,
getId:getId,
title:title};exports.default=_default;
