// rnv/dist/platformTools/web/webNext.js
var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.exportWebNext=exports.deployWebNext=exports.runWebDevServer=exports.buildWebNext=exports.runWebNext=void 0;var _defineProperty2=_interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));var _regenerator=_interopRequireDefault(require("@babel/runtime/regenerator"));
var _path=_interopRequireDefault(require("path"));
var _fs=_interopRequireDefault(require("fs"));
var _chalk=_interopRequireDefault(require("chalk"));
var _betterOpn=_interopRequireDefault(require("better-opn"));
var _exec=require("../../systemTools/exec");
var _common=require("../../common");





var _logger=require("../../systemTools/logger");
var _constants=require("../../constants");
var _webTools=require("../../deployTools/webTools");
var _fileutils=require("../../systemTools/fileutils");function ownKeys(object,enumerableOnly){var keys=Object.keys(object);if(Object.getOwnPropertySymbols){var symbols=Object.getOwnPropertySymbols(object);if(enumerableOnly)symbols=symbols.filter(function(sym){return Object.getOwnPropertyDescriptor(object,sym).enumerable;});keys.push.apply(keys,symbols);}return keys;}function _objectSpread(target){for(var i=1;i<arguments.length;i++){var source=arguments[i]!=null?arguments[i]:{};if(i%2){ownKeys(Object(source),true).forEach(function(key){(0,_defineProperty2.default)(target,key,source[key]);});}else if(Object.getOwnPropertyDescriptors){Object.defineProperties(target,Object.getOwnPropertyDescriptors(source));}else{ownKeys(Object(source)).forEach(function(key){Object.defineProperty(target,key,Object.getOwnPropertyDescriptor(source,key));});}}return target;}

var configureNextIfRequired=function configureNextIfRequired(c){var _c$paths$appConfig$fo;var _c$paths$project,platformTemplatesDirs,dir,publicDir,baseFontsDir,stylesDir,pagesDir,_appFile,platformTemplateDir,configFile,fontsSymLinkPath,cssOutput,fontFiles;return _regenerator.default.async(function configureNextIfRequired$(_context){while(1){switch(_context.prev=_context.next){case 0:_c$paths$project=
c.paths.project,platformTemplatesDirs=_c$paths$project.platformTemplatesDirs,dir=_c$paths$project.dir;
publicDir=_path.default.join(dir,'public');
baseFontsDir=(_c$paths$appConfig$fo=c.paths.appConfig.fontDirs)==null?void 0:_c$paths$appConfig$fo[0];
stylesDir=_path.default.join(dir,'styles');
pagesDir=_path.default.resolve((0,_common.getConfigProp)(c,c.platform,'pagesDir')||'src/app');
_appFile=_path.default.join(pagesDir,'_app.js');
platformTemplateDir=_path.default.join(platformTemplatesDirs[c.platform],"_"+c.platform);
configFile=_path.default.join(dir,_constants.NEXT_CONFIG_NAME);


!_fs.default.existsSync(publicDir)&&_fs.default.mkdirSync(publicDir);
fontsSymLinkPath=_path.default.join(publicDir,'fonts');

if(_fs.default.existsSync(baseFontsDir)){
if(!_fs.default.existsSync(fontsSymLinkPath)){
try{
_fs.default.unlinkSync(fontsSymLinkPath);
}catch(e){
(0,_logger.logDebug)(e);
}
_fs.default.symlinkSync(baseFontsDir,fontsSymLinkPath);
}


if(!_fs.default.existsSync(stylesDir)){
_fs.default.mkdirSync(stylesDir);
cssOutput='';

fontFiles=_fs.default.readdirSync(baseFontsDir);
fontFiles.forEach(function(file){
cssOutput+="\n                  @font-face {\n                      font-family: '"+

file.split('.')[0]+"';\n                      src: url('/fonts/"+
file+"');\n                  }\n\n              ";



});

(0,_fileutils.fsWriteFileSync)(_path.default.join(stylesDir,'fonts.css'),cssOutput);
}
}



if(!_fs.default.existsSync(_appFile)){
if(!_fs.default.existsSync(pagesDir)){
_fs.default.mkdirSync(pagesDir);
}
(0,_fileutils.writeCleanFile)(_path.default.join(platformTemplateDir,'_app.js'),_appFile,[{pattern:'{{FONTS_CSS}}',override:_path.default.relative(pagesDir,_path.default.resolve('styles/fonts.css')).replace(/\\/g,'/')}],null,c);
}


if(!_fs.default.existsSync(configFile)){
(0,_fileutils.writeCleanFile)(_path.default.join(platformTemplateDir,_constants.NEXT_CONFIG_NAME),configFile,null,null,c);
}case 13:case"end":return _context.stop();}}},null,null,null,Promise);};


var runWebNext=function runWebNext(c,platform,port){var devServerHost,isPortActive;return _regenerator.default.async(function runWebNext$(_context2){while(1){switch(_context2.prev=_context2.next){case 0:
(0,_logger.logTask)("runWebNext:"+platform+":"+port);_context2.next=3;return _regenerator.default.awrap(
configureNextIfRequired(c));case 3:

devServerHost=c.runtime.localhost;_context2.next=6;return _regenerator.default.awrap(

(0,_common.checkPortInUse)(c,platform,port));case 6:isPortActive=_context2.sent;if(

isPortActive){_context2.next=15;break;}
(0,_logger.logInfo)("Looks like your "+
_chalk.default.white(platform)+" devServerHost "+_chalk.default.white(devServerHost)+" at port "+_chalk.default.white(
port)+" is not running. Starting it up for you...");_context2.next=11;return _regenerator.default.awrap(


_runWebBrowser(c,platform,devServerHost,port,false));case 11:_context2.next=13;return _regenerator.default.awrap(
runWebDevServer(c,platform,port));case 13:_context2.next=19;break;case 15:_context2.next=17;return _regenerator.default.awrap(

(0,_common.confirmActiveBundler)(c));case 17:_context2.next=19;return _regenerator.default.awrap(
_runWebBrowser(c,platform,devServerHost,port,true));case 19:case"end":return _context2.stop();}}},null,null,null,Promise);};exports.runWebNext=runWebNext;



var _runWebBrowser=function _runWebBrowser(c,platform,devServerHost,port,alreadyStarted){return new Promise(function(resolve){
(0,_logger.logTask)("_runWebBrowser:"+platform+":"+devServerHost+":"+port+":"+c.runtime.shouldOpenBrowser);
if(!c.runtime.shouldOpenBrowser)return resolve();
var wait=(0,_common.waitForWebpack)(c,'next').
then(function(){
(0,_betterOpn.default)("http://"+devServerHost+":"+port+"/");
}).
catch(function(e){
(0,_logger.logWarning)(e);
});
if(alreadyStarted)return wait;
return resolve();
});};

var exportNext=function exportNext(c){var env,pagesDir;return _regenerator.default.async(function exportNext$(_context3){while(1){switch(_context3.prev=_context3.next){case 0:
(0,_logger.logTask)('exportWebNext');_context3.next=3;return _regenerator.default.awrap(
configureNextIfRequired(c));case 3:
env=(0,_common.getConfigProp)(c,c.platform,'environment');
pagesDir=(0,_common.getConfigProp)(c,c.platform,'pagesDir');
if(!pagesDir)(0,_logger.logWarning)("You're missing "+c.platform+".pagesDir config. Defaulting to 'src/app'");return _context3.abrupt("return",

(0,_exec.executeAsync)(c,"npx next export ./platformBuilds/"+c.runtime.appId+"_"+c.platform+" --pagesDir "+(pagesDir||'src/app'),_objectSpread(_objectSpread({},process.env),{},{env:{NODE_ENV:env||'development'},interactive:true})));case 7:case"end":return _context3.stop();}}},null,null,null,Promise);};


var buildWebNext=function buildWebNext(c){var env,pagesDir;return _regenerator.default.async(function buildWebNext$(_context4){while(1){switch(_context4.prev=_context4.next){case 0:
(0,_logger.logTask)('buildWebNext');_context4.next=3;return _regenerator.default.awrap(
configureNextIfRequired(c));case 3:
env=(0,_common.getConfigProp)(c,c.platform,'environment');
pagesDir=(0,_common.getConfigProp)(c,c.platform,'pagesDir');
if(!pagesDir)(0,_logger.logWarning)("You're missing "+c.platform+".pagesDir config. Defaulting to 'src/app'");_context4.next=8;return _regenerator.default.awrap(

(0,_exec.executeAsync)(c,"npx next build ./platformBuilds/"+c.runtime.appId+"_"+c.platform+" --pagesDir "+(pagesDir||'src/app'),_objectSpread(_objectSpread({},process.env),{},{env:{NODE_ENV:env||'development'},interactive:true})));case 8:return _context4.abrupt("return",
exportNext(c));case 9:case"end":return _context4.stop();}}},null,null,null,Promise);};exports.buildWebNext=buildWebNext;


var runWebDevServer=function runWebDevServer(c,platform,port){
(0,_logger.logTask)("runWebDevServer:"+platform);
var env=(0,_common.getConfigProp)(c,platform,'environment');
var pagesDir=(0,_common.getConfigProp)(c,platform,'pagesDir');
if(!pagesDir)(0,_logger.logWarning)("You're missing "+platform+".pagesDir config. Defaulting to 'src/app'");

return(0,_exec.executeAsync)(c,"npx next --pagesDir "+(pagesDir||'src/app')+" --port "+port,{env:{NODE_ENV:env||'development'},interactive:true});
};exports.runWebDevServer=runWebDevServer;

var deployWebNext=function deployWebNext(c,platform){
(0,_logger.logTask)("deployWeb:"+platform);
return(0,_webTools.selectWebToolAndDeploy)(c,platform);
};exports.deployWebNext=deployWebNext;

var exportWebNext=function exportWebNext(c,platform){
(0,_logger.logTask)("exportWeb:"+platform);
return(0,_webTools.selectWebToolAndExport)(c,platform);
};exports.exportWebNext=exportWebNext;
